kubernetes/deploy.sh

```diff
@@ -49,4 +122,10 @@ if [[ -z $CLUSTER_DNS_IP ]]; then
   fi
 fi
 
-sed -e s/CLUSTER_DNS_IP/$CLUSTER_DNS_IP/g -e s/CLUSTER_DOMAIN/$CLUSTER_DOMAIN/g -e "s?REVERSE_CIDRS?$REVERSE_CIDRS?g" $YAML_TEMPLATE
+if [[ "${SKIP}" -ne 1 ]] ; then
+    translate-kube-dns-configmap
+fi
+
+orig=$'\n'
+replace=$'\\\n'
+sed -e "s/CLUSTER_DNS_IP/$CLUSTER_DNS_IP/g" -e "s/CLUSTER_DOMAIN/$CLUSTER_DOMAIN/g" -e "s?REVERSE_CIDRS?$REVERSE_CIDRS?g" -e "s@STUBDOMAINS@${STUBDOMAINS//$orig/$replace}@g" -e "s@FEDERATIONS@${FEDERATIONS//$orig/$replace}@g" -e "s/UPSTREAMNAMESERVER/$UPSTREAM/g" ${YAML_TEMPLATE}
```

Comment:
Is there a better way to process multiline variables in sed?

Reply:
You can break a line up into multiple lines to make it more readable.... e.g.

```
sed -e "s/CLUSTER_DNS_IP/$CLUSTER_DNS_IP/g" \ 
    -e "s/CLUSTER_DOMAIN/$CLUSTER_DOMAIN/g" \
    -e "s?REVERSE_CIDRS?$REVERSE_CIDRS?g" \
    -e "s@STUBDOMAINS@${STUBDOMAINS//$orig/$replace}@g" \
    -e "s@FEDERATIONS@${FEDERATIONS//$orig/$replace}@g" \
    -e "s/UPSTREAMNAMESERVER/$UPSTREAM/g" \
    ${YAML_TEMPLATE}
```